src/cpp/session/modules/SessionFind.cpp

```diff
@@ -987,26 +987,93 @@ class GrepOperation : public boost::enable_shared_from_this<GrepOperation>
 
 } // namespace
 
-struct GrepOptions
+class GrepOptions : public boost::noncopyable
 {
+public:
+
    GrepOptions(std::string search, std::string directory,
       json::Array includeFilePatterns, json::Array excludeFilePatterns,
       bool asRegex, bool ignoreCase) :
-      asRegex(asRegex),
-      ignoreCase(ignoreCase),
-      searchPattern(search),
-      directory(directory),
-      includeFilePatterns(includeFilePatterns),
-      excludeFilePatterns(excludeFilePatterns)
-   {}
+      asRegex_(asRegex),
+      ignoreCase_(ignoreCase),
+      searchPattern_(search),
+      directory_(directory),
+      includeFilePatterns_(includeFilePatterns),
+      excludeFilePatterns_(excludeFilePatterns)
+   {
+      processExcludeFilePatterns();
+   }
+
+   const bool asRegex() const
+   {
+      return asRegex_;
+   }
 
-   bool asRegex;
-   bool ignoreCase;
+   const bool ignoreCase() const
+   {
+      return ignoreCase_;
+   }
+
+   const std::string& searchPattern() const
+   {
+      return searchPattern_;
+   }
+
+   const std::string& directory() const
+   {
+      return directory_;
+   }
+
+   const json::Array& includeFilePatterns() const
+   {
+      return includeFilePatterns_;
+   }
+
+   const json::Array& excludeFilePatterns() const
+   {
+      return excludeFilePatterns_;
+   }
+
+   const bool gitFlag() const
```

Comment:
Nit: no need to return a const bool since you're returning by-value rather than by-reference.

Reply:
fixed